pkg/bitwarden/cli.go

```diff
@@ -142,3 +144,170 @@ func (c *cliClient) GetPassword(itemName string) ([]byte, error) {
 func (c *cliClient) Logout() ([]byte, error) {
 	return c.run("logout")
 }
+
+func createItem(itemTemplate string, targetItem *Item, c *cliClient) error {
+	// the bitwarden cli expects the item to be base64 encoded
+	encItem := b64.StdEncoding.EncodeToString([]byte(itemTemplate))
+	out, err := c.run("--session", c.session, "create", "item", encItem)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(out, targetItem)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createAttachment(fileContents []byte, fileName string, itemID string, newAttachment *Attachment, c *cliClient) error {
+	filePath := "/tmp/" + fileName
+	if err := ioutil.WriteFile(filePath, fileContents, 0644); err != nil {
+		return fmt.Errorf("failed to create temporary file for new attachment: %w", err)
+	}
+	out, err := c.run("--session", c.session, "create", "attachment", "--file", filePath, "--itemid", itemID)
+	if err != nil {
+		return fmt.Errorf("bw create failed: %w", err)
+	}
+	if err = json.Unmarshal(out, newAttachment); err != nil {
+		return fmt.Errorf("failed to parse bw output: %w", err)
+	}
+
+	return nil
+}
+
+func createEmptyItem(itemName string, targetItem *Item, c *cliClient) error {
+	itemTemplate := fmt.Sprintf(`{"type":1,"name":"%s","login":{}}`, itemName)
+	return createItem(itemTemplate, targetItem, c)
+}
+
+func createItemWithPassword(itemName string, password []byte, targetItem *Item, c *cliClient) error {
+	itemTemplate := fmt.Sprintf(`{"type":1,"name":"%s","login":{"password":"%s"}}`, itemName, string(password))
+	return createItem(itemTemplate, targetItem, c)
+}
```

Comment:
Quotes are usually valid characters in passwords, we should go through the `json` library instead of using string formatting to be safe.